src/parser.rs

```diff
@@ -3438,173 +3438,199 @@ impl<'a> Parser<'a> {
     }
 
     pub fn parse_alter(&mut self) -> Result<Statement, ParserError> {
-        self.expect_keyword(Keyword::TABLE)?;
-        let _ = self.parse_keyword(Keyword::ONLY);
-        let table_name = self.parse_object_name()?;
-        let operation = if self.parse_keyword(Keyword::ADD) {
-            if let Some(constraint) = self.parse_optional_table_constraint()? {
-                AlterTableOperation::AddConstraint(constraint)
-            } else {
-                let if_not_exists =
-                    self.parse_keywords(&[Keyword::IF, Keyword::NOT, Keyword::EXISTS]);
-                if self.parse_keyword(Keyword::PARTITION) {
+        let object_type = self.expect_one_of_keywords(&[Keyword::TABLE, Keyword::INDEX])?;
```

Comment:
reviewing with https://github.com/sqlparser-rs/sqlparser-rs/pull/767/files?w=1 ignoring whitespace made the changes much easier for me to see